cfme/cloud/security_group.py

```diff
@@ -0,0 +1,264 @@
+    name = TextInput(name='name')
+    description = TextInput(name='description')
+    cloud_tenant = Select(name='cloud_tenant_id')
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+
+class SecurityGroupView(BaseLoggedInPage):
+    """Base view for header and nav checking, navigatable views should inherit this"""
+
+    flash = FlashMessages('.//div[@id="flash_msg_div"]'
+                          '/div[@id="flash_text_div" or contains(@class, "flash_text_div")]')
+
+    @property
+    def in_security_groups(self):
+        return(
+            self.logged_in_as_current_user and
+            self.navigation.currently_selected == ['Networks', 'Security Groups'])
+
+
+class SecurityGroupAllView(SecurityGroupView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_security_groups and
+            self.entities.title.text == 'Security Groups')
+
+    toolbar = View.nested(SecurityGroupToolbar)
+    including_entities = View.include(BaseEntitiesView, use_parent=True)
+
+
+class SecurityGroupDetailsView(SecurityGroupView):
+    @property
+    def is_displayed(self):
+        expected_title = '{} (Summary)'.format(self.context['object'].name)
+        return (
+            self.in_security_groups and
+            self.entities.title.text == expected_title and
+            self.entities.breadcrumb.active_location == expected_title)
+
+    toolbar = View.nested(SecurityGroupDetailsToolbar)
+    sidebar = View.nested(SecurityGroupDetailsAccordion)
+    entities = View.nested(SecurityGroupDetailsEntities)
+
+
+class SecurityGroupAddView(SecurityGroupView):
+    @property
+    def is_displayed(self):
+        return (
+            self.in_security_groups and
+            self.entities.breadcrumb.active_location == 'Add New Security Group' and
+            self.entities.title.text == 'Add New Security Group')
+
+    entities = View.nested(SecurityGroupAddEntities)
+    form = View.nested(SecurityGroupAddForm)
+
+
+@attr.s
+class SecurityGroup(BaseEntity):
+    """ Automate Model page of SecurityGroups
+
+    Args:
+        provider (obj): Provider name for Network Manager
+        name(str): name of the Security Group
+        description (str): Security Group description
+    """
+    _param_name = "SecurityGroup"
+
+    name = attr.ib()
+    provider = attr.ib()
+    description = attr.ib(default="")
+
+    def refresh(self):
+        self.provider.refresh_provider_relationships()
+        self.browser.refresh()
+
+    def delete(self, cancel=False, wait=False):
+        view = navigate_to(self, 'Details')
+        view.toolbar.configuration.item_select('Delete this Security Group',
+                                               handle_alert=(not cancel))
+        # cancel doesn't redirect, confirmation does
+        view.flush_widget_cache()
+        if not cancel:
+            view = navigate_to(self.parent, 'All')
+            # view = self.create_view(SecurityGroupAllView)
```

Comment:
**Required**

I think the right thing to do here is create the view, not navigate.  In the case where the delete is confirmed, we want to assert that the redirect happened, and the flash message is displayed.

Remove the navigate_to, un-comment the `create_view` call, and then assert view.is_displayed before calling assert_success_message.